BSPConversionLib/Source/EntityConverter.cs

```diff
@@ -551,6 +551,50 @@ private void GiveWeaponOnStartTouch(Entity trigger, Entity weaponEnt)
 				fireOnce = -1
 			};
 			trigger.connections.Add(connection);
+
+			if (weaponEnt["count"] == "-1")
+				return;
+
+			if (!weaponEnt.TryGetValue("count", out _) || weaponEnt["count"] == "0") // Every quake weapon has a default ammo count when none is specified
```

Comment:
You should call `weaponEnt.TryGetValue("count", out var count)` before line 555 and use the `count` variable instead of weaponEnt["count"] in your if conditions.